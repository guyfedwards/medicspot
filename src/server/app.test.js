const supertest = require('supertest')
const createApp = require('./app')
const db = require('./db')

let app
let database

describe('/locations', () => {
  beforeAll(async () => {
    database = await db.init()
    app = createApp(database)
  })

  it('should return empty array if the query is shorter than 2 characters', async () => {
    const res = await supertest(app)
      .get('/locations?q=s')

    expect(res.body.data.length).toBe(0)
  })

  it('should match full name', async () => {
    const res = await supertest(app)
      .get('/locations?q=phrakhtaes')

    expect(res.body.data.length).not.toBe(0)
    expect(res.body.data).toMatchSnapshot()
  })

  it('should match a partial name', async () => {
    const res = await supertest(app)
      .get('/locations?q=hastin')

    expect(res.body.data.length).not.toBe(0)
    expect(res.body.data).toMatchSnapshot()
  })
})
