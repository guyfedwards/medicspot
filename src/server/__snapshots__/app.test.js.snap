// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/locations should match a partial name 1`] = `
Array [
  Object {
    "geonameid": "2633418",
    "latitude": "52.2055",
    "longitude": "-0.73488",
    "name": "Yardley Hastings",
  },
  Object {
    "geonameid": "2634582",
    "latitude": "52.18333",
    "longitude": "-1.58333",
    "name": "Wellesbourne Hastings",
  },
  Object {
    "geonameid": "6287208",
    "latitude": "50.8559",
    "longitude": "0.58506",
    "name": "Hastings Castle",
  },
  Object {
    "geonameid": "6506607",
    "latitude": "53.0084",
    "longitude": "-2.2294",
    "name": "Hastings Slieve Donard Hotel",
  },
  Object {
    "geonameid": "6943685",
    "latitude": "54.98203",
    "longitude": "-7.33573",
    "name": "Hastings Everglades Hotel",
  },
  Object {
    "geonameid": "6945176",
    "latitude": "50.858",
    "longitude": "0.576",
    "name": "Hastings Railway Station",
  },
  Object {
    "geonameid": "6945193",
    "latitude": "54.6512",
    "longitude": "-5.80218",
    "name": "Hastings Culloden Estate",
  },
  Object {
    "geonameid": "6945194",
    "latitude": "54.5908",
    "longitude": "-5.93448",
    "name": "Hastings Europa Hotel",
  },
  Object {
    "geonameid": "7290687",
    "latitude": "50.86728",
    "longitude": "0.58145",
    "name": "Hastings District",
  },
  Object {
    "geonameid": "7290987",
    "latitude": "52.1941",
    "longitude": "-0.74418",
    "name": "Yardley Hastings",
  },
  Object {
    "geonameid": "7292142",
    "latitude": "51.67714",
    "longitude": "-1.62772",
    "name": "Eaton Hastings",
  },
  Object {
    "geonameid": "7292678",
    "latitude": "52.2928",
    "longitude": "-1.32326",
    "name": "Leamington Hastings",
  },
  Object {
    "geonameid": "7293959",
    "latitude": "52.50188",
    "longitude": "-1.38583",
    "name": "Burton Hastings",
  },
  Object {
    "geonameid": "7295280",
    "latitude": "51.16348",
    "longitude": "0.99241",
    "name": "Hastingleigh",
  },
  Object {
    "geonameid": "8299274",
    "latitude": "51.7476",
    "longitude": "0.15115",
    "name": "Hastingwood",
  },
  Object {
    "geonameid": "9951116",
    "latitude": "54.59439",
    "longitude": "-5.84115",
    "name": "Hastings Stormont",
  },
  Object {
    "geonameid": "10106731",
    "latitude": "54.89917",
    "longitude": "-5.86104",
    "name": "Hastings Ballygally Castle Hotel",
  },
  Object {
    "geonameid": "10106738",
    "latitude": "54.6511",
    "longitude": "-5.80447",
    "name": "Hastings Culloden Estate & Spa",
  },
  Object {
    "geonameid": "10243797",
    "latitude": "54.21571",
    "longitude": "-5.88477",
    "name": "Hastings Slieve Donard Resort And Spa",
  },
]
`;

exports[`/locations should match full name 1`] = `
Array [
  Object {
    "geonameid": "5653",
    "latitude": "34.56717",
    "longitude": "33.02724",
    "name": "Phrakhtaes",
  },
]
`;
